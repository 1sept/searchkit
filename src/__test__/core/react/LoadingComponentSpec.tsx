import * as React from "react";

import {
  SearchkitManager,
  LoadingComponent
} from "../../../"

import {mount} from "enzyme";

describe("SearchkitProvider", ()=> {

  beforeEach(()=> {
    this.searchkit = SearchkitManager.mock()
    this.wrapper = mount(
      <LoadingComponent searchkit={this.searchkit}>
        <p>loading...</p>
      </LoadingComponent>
    )
  })

  test("display nothing when not loading", ()=> {
    expect(this.wrapper.html()).toBe("<div></div>")
  })

  test("display children when loading", ()=> {
    this.searchkit.loading = true
    this.wrapper.update()
    expect(this.wrapper.html()).toBe("<p>loading...</p>")
  })


})
