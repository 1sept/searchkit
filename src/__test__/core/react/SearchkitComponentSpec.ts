import {
  SearchkitComponent
} from "../../../"

describe("SearchkitComponent", ()=> {

  beforeEach(()=> {

  })

  it("SearchkitComponent.translatePropType", ()=> {
    let translations = {
      continueButton:"Continue",
      cancelButton:"Cancel"
    }

    let handler = SearchkitComponent
      .translationsPropType(translations)

    expect(handler(
      {translations:{
        continueButton:"Continue..."
      }},
      "translations", "MyComponent"
    )).toEqual(null)

    expect(handler(
      {translations:{
        unknown1:"",
        unknown2:""
      }},
      "translations", "MyComponent"
    )).toEqual(new Error(
      "MyComponent: incorrect translations, unknown1,unknown2 keys are not included in continueButton,cancelButton"
    ))


  })


})
