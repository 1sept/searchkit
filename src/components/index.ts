export * from "./search/filters/menu-filter/src/MenuFilter";
export * from "./search/filters/refinement-list-filter/src/RefinementListFilter";
export * from "./search/filters/reset-filters/src/ResetFilters";
export * from "./search/filters/selected-filters/src/SelectedFilters";
// export * from "./search/filters/hierarchical-menu-filter/src/HierarchicalMenuFilter";
export * from "./search/hits/src/Hits";
export * from "./search/hits-stats/src/HitsStats";
export * from "./search/pagination/src/Pagination";
export * from "./search/search-box/src/SearchBox";
// export {default as SearchkitComponent} from "./SearchkitComponent";
// export {default as SearchkitProvider} from "./SearchkitProvider";
