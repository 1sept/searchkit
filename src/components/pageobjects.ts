export * from "./search/filters/facet-filter/pageobjects"
export * from "./search/hits/page-objects/Hits"
export * from "./search/hits-stats/page-objects/HitsStats"
export * from "./search/search-box/page-objects/Searchbox"
export * from "./search/search-box/page-objects/Loader"
export * from "./ui/list/pageobjects"
