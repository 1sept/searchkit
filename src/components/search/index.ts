export {default as MenuFilter} from "./filters/menu-filter/src/MenuFilter.tsx";
export {default as RefinementListFilter} from "./filters/refinement-list-filter/src/RefinementListFilter.tsx";
export {default as ResetFilters} from "./filters/reset-filters/src/ResetFilters.tsx";
export {default as SelectedFilters} from "./filters/selected-filters/src/SelectedFilters.tsx";
export {default as HierarchicalMenuFilter} from "./filters/hierarchical-menu-filter/src/HierarchicalMenuFilter.tsx";
export {default as Hits} from "./hits/src/Hits.tsx";
export {default as HitsStats} from "./hits-stats/src/HitsStats.tsx";
export {default as Pagination} from "./pagination/src/Pagination.tsx";
export {default as SearchBox} from "./search-box/src/SearchBox.tsx";
