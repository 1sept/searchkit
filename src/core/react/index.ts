export * from "./SearchkitComponent"
export * from "./SearchkitProvider"
export * from "./LoadingComponent"
export * from "./ReactComponentType"
export * from "./FastClick"
export * from "./RenderComponent"
export * from "./pure-render/index"
