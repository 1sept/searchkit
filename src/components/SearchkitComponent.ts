import * as React from "react"
import ESClient from "../domain/ESClient.ts";
import Accessor from "../domain/accessors/Accessor.ts";

export default class SearchkitComponent<P,S> extends React.Component<P,S> {
  searcher:ESClient
  accessor:Accessor

	static contextTypes = {
		searcher:React.PropTypes.instanceOf(ESClient)    
	}
  defineAccessor():Accessor{
    return null
  }
  componentWillMount(){
    this.searcher = this.context["searcher"]
    this.accessor  = this.defineAccessor()
    if(this.accessor){
      this.searcher.stateManager.registerAccessor(this.accessor)
    }
  }
}
