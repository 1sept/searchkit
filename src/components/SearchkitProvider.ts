import * as React from "react";
import ESClient from "../domain/ESClient.ts"
import * as Rx from "rx"

export interface ISearcherProvider {
	searcher:ESClient
	children?:any
}

export default class SearchkitProvider extends React.Component<ISearcherProvider,any> {
	static childContextTypes = {
		searcher:React.PropTypes.instanceOf(ESClient)
	}

	private searcher: ESClient;
	results:any
	searcherUnsubscribe:Rx.IDisposable

	constructor(props:ISearcherProvider){
		super(props)
		this.searcher = props.searcher
	}

	componentWillMount(){
		this.searcherUnsubscribe = this.searcher.resultsListener.subscribe(
			()=> this.forceUpdate()
		)
	}
	componentDidMount(){
		this.searcher.completeRegistration()
	}

	componentWillUnmount(){
		this.searcherUnsubscribe.dispose()
	}

	getChildContext(){
		return {searcher:this.props.searcher}
	}

	render(){
		return this.props.children
	}
}
