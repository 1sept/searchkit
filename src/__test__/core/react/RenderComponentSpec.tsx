import * as React from "react";
import {mount} from "enzyme";
import * as ReactTestRenderer from 'react-test-renderer'

import {
  fastClick, hasClass, jsxToHTML, printPrettyHtml
} from "../../../components/__test__/TestHelpers"

import {Panel} from "../../../components"

import {
  renderComponent,
  RenderComponentType,
  RenderComponentPropType
} from "../../../"


fdescribe("RenderComponent", ()=> {

  beforeEach(()=> {
    this.SubPanel = class SubPanel extends Panel {
    }
    this.SubPanel.defaultProps.title = "SubPanel"
    this.SubPanelElement = <Panel title="PanelElement"/>
    this.PanelReactClass = React.createClass({
      contextTypes:{
        color:React.PropTypes.string
      },
      render(){
        return (<Panel title={"PanelReactClass " + this.context.color} {...this.props}>
          {this.props.children}
        </Panel>)
      }
    })

    this.PanelFunction = (props, context)=> {
      return (
        <Panel title={"PanelFunction " + context.color}>
          {props.children}
        </Panel>
      )
    }
    this.PanelFunction.contextTypes = {
      color:React.PropTypes.string
    }

    class Provider extends React.Component<any, any>{
      static childContextTypes = {
        color: React.PropTypes.string
      }
      getChildContext(){
        return {color:"purple"}
      }
      render(){
        return this.props.children
      }
    }

    this.mount = (component, props={})=> {
      return ReactTestRenderer.create(
        <Provider>
          {renderComponent(
            component, props,
            <p>content..</p>
          )}
        </Provider>
      ).toJSON()
    }

  })

  test("React.Component class", ()=> {
    expect(this.mount(this.SubPanel)).toMatchSnapshot()
  })

  test("React Element", ()=> {
    expect(this.mount(this.SubPanelElement)).toMatchSnapshot()
  })

  test("React Class", ()=> {
    expect(this.mount(this.PanelReactClass)).toMatchSnapshot()
  })

  test("Render function", ()=> {
    expect(this.mount(this.PanelFunction)).toMatchSnapshot()
  })

  // xtest("Invalid component", ()=> {
  //   spyOn(console, "warn")
  //   try{
  //     this.mount(10)
  //     // printPrettyHtml(this.wrapper.html())
  //   } catch (e){
  //
  //   }
  //   expect(console.warn).toHaveBeenCalledWith(
  //     "Invalid component", 10
  //   )
  //
  // })

})
