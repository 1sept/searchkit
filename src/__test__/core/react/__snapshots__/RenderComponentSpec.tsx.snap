// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderComponent React Class 1`] = `
<div
  className="sk-panel"
>
  <div
    className="sk-panel__header"
  >
    PanelReactClass purple
  </div>
  <div
    className="sk-panel__content"
  >
    <p>
      content..
    </p>
  </div>
</div>
`;

exports[`RenderComponent React Element 1`] = `
<div
  className="sk-panel"
>
  <div
    className="sk-panel__header"
  >
    PanelElement
  </div>
  <div
    className="sk-panel__content"
  >
    <p>
      content..
    </p>
  </div>
</div>
`;

exports[`RenderComponent React.Component class 1`] = `
<div
  className="sk-panel"
>
  <div
    className="sk-panel__header"
  >
    SubPanel
  </div>
  <div
    className="sk-panel__content"
  >
    <p>
      content..
    </p>
  </div>
</div>
`;

exports[`RenderComponent Render function 1`] = `
<div
  className="sk-panel"
>
  <div
    className="sk-panel__header"
  >
    PanelFunction purple
  </div>
  <div
    className="sk-panel__content"
  >
    <p>
      content..
    </p>
  </div>
</div>
`;
